### ISSUE TITLE
fake_processor doesn't support promotion_code

### ISSUE THREAD

**Issue by dwarfmondo** [pay-rails/pay#790]
I'm seeing an issue where it appears that `fake_processor` doesn't support the `promotion_code` parameter. 

In my tests, the following returns a `NoMethodError`…

```ruby
@account.payment_processor.subscribe(plan: @plan.token, trial_period_days: @plan.trial_period_days, promotion_code: "promo_1234")
```
returns…
```
NoMethodError: undefined method `processor' for nil:NilClass

      @payment_processor ||= self.class.pay_processor_for(customer.processor).new(self)
                                                                  ^^^^^^^^^^
```

Is this a known issue? Is there a work around?

Thanks for the help!

**Comment by excid3** [comment]
Correct, it does not. Only Stripe supports that parameter.

I'd accept a PR that allows arbitrary arguments for the fake processor subscriptions. It would just need to ignore them.

**Comment by dwarfmondo** [comment]
@excid3 Gotcha. I think I can knock that out. I also noticed that `Pay::FakeProcessor::Billable#update_customer!` also doesn't support arbitrary arguments. Should I include that change as well?

**Comment by excid3** [comment]
Yeah sure. 👍  I'm guessing there will be a good handful of places that could use updates.

**Comment by dwarfmondo** [comment]
@excid3 Sorry for the delay. Our kids spring break was this past week. Here's the [PR](https://github.com/pay-rails/pay/pull/795) that adds support for promotion codes with `fake_processor`. Let me know next steps. Happy to help out. 

Thanks!

**Comment by dwarfmondo** [comment]
@excid3 Just wanted to follow-up on this. Any chance you've had a look and have feedback? Thanks!

**Comment by excid3** [comment]
@dwarfmondo I haven't yet, my son and I have both been out sick so it's been a rough couple weeks. Will take a look shortly!

**Comment by excid3** [comment]
Closed by #795
